nomos-blend/network/src/core/with_core/behaviour/tests/utils.rs

```diff
@@ -88,7 +112,20 @@ impl BehaviourBuilder {
         self
     }
 
-    pub fn build(self) -> Behaviour<AlwaysTrueVerifier, IntervalProvider> {
+    pub fn with_session(mut self, session: SessionNumber) -> Self {
+        assert!(self.session.is_none(), "session already set.");
+        self.session = Some(session);
+        self
+    }
+}
+
+impl<ProofsVerifier> BehaviourBuilder<ProofsVerifier>
+where
+    ProofsVerifier: encap::ProofsVerifier,
+{
+    pub fn build(self) -> Behaviour<ProofsVerifier, IntervalProvider> {
```

Comment:
I think we can specify the proof verifier generic here instead of having it in the struct and basically never use it. What do you think?

Reply:
Cause we use it as phantom data, so I am not sure what we use use it for other than returning a behaviour that contains it, so a generic on this function should be enough.